Framework/CurveFitting/src/FitMW.cpp

```diff
@@ -162,16 +163,16 @@ void FitMW::createDomain(boost::shared_ptr<API::FunctionDomain> &domain,
                          size_t i0) {
   setParameters();
 
-  const Mantid::MantidVec &X = m_matrixWorkspace->readX(m_workspaceIndex);
+  auto X = m_matrixWorkspace->points(m_workspaceIndex);
 
-  if (X.empty()) {
+  if (X.size() == 0) {
```

Comment:
I think I will add an `empty()` method, such that our new types are more compliant with `std::vector`.